src/Neo/Cryptography/Crypto.cs

```diff
@@ -48,20 +50,40 @@ public static byte[] Hash256(ReadOnlySpan<byte> message)
         /// Signs the specified message using the ECDSA algorithm.
         /// </summary>
         /// <param name="message">The message to be signed.</param>
-        /// <param name="prikey">The private key to be used.</param>
-        /// <param name="pubkey">The public key to be used.</param>
+        /// <param name="priKey">The private key to be used.</param>
+        /// <param name="ecCurve">The <see cref="ECC.ECCurve"/> curve of the signature, default is <see cref="ECC.ECCurve.Secp256k1"/>.</param>
         /// <returns>The ECDSA signature for the specified message.</returns>
-        public static byte[] Sign(byte[] message, byte[] prikey, byte[] pubkey)
+        public static byte[] Sign(byte[] message, byte[] priKey, ECC.ECCurve ecCurve = null)
         {
+            if (IsOSX && ecCurve == ECC.ECCurve.Secp256k1)
+            {
+                var curveParameters = Org.BouncyCastle.Asn1.Sec.SecNamedCurves.GetByName("secp256k1");
+                var domain = new ECDomainParameters(curveParameters.Curve, curveParameters.G, curveParameters.N, curveParameters.H);
+                var signer = new Org.BouncyCastle.Crypto.Signers.ECDsaSigner();
+                var privateKey = new BigInteger(1, priKey);
+                var priKeyParameters = new ECPrivateKeyParameters(privateKey, domain);
+                signer.Init(true, priKeyParameters);
+                var signature = signer.GenerateSignature(message.Sha256());
+
+                var signatureBytes = new byte[64];
+                var rBytes = signature[0].ToByteArrayUnsigned();
+                var sBytes = signature[1].ToByteArrayUnsigned();
+
+                // Copy r and s into their respective parts of the signatureBytes array, aligning them to the right.
+                Buffer.BlockCopy(rBytes, 0, signatureBytes, 32 - rBytes.Length, rBytes.Length);
+                Buffer.BlockCopy(sBytes, 0, signatureBytes, 64 - sBytes.Length, sBytes.Length);
+                return signatureBytes;
+            }
+
+            var curve =
+                ecCurve == null || ecCurve == ECC.ECCurve.Secp256r1 ? ECCurve.NamedCurves.nistP256 :
+                ecCurve == ECC.ECCurve.Secp256k1 ? ECCurve.CreateFromFriendlyName("secP256k1") :
```

Comment:
How about making the `ECCurve.CreateFromFriendlyName("secP256k1")` a `public static` property to allocate this object once, and then reuse it everywhere?

Reply:
maybe a cache system in another pr